# Panalysis Business Analytics Case Study

**Author** Indraja Bandla


# 3. How many subscribers are subscribed to more than one list?

In [103]:
data.groupby(['SubscriberID','year']).ListID.value_counts()

SubscriberID                                                      year  ListID                     
000079b501e0e82b7ef2addf1db8b860f6e3cb125d3b0b92fdb683f689cb5e68  2015  Balkan News                    1
0001c6172c43da3dde25b87e10c23861250aed7ca2ce3c4d8eae1e41f51ef600  2015  Fishing with Friends           1
0001cec76add203c3fea01001ed2729b0ddffaad9658b027307fe1932acd8771  2014  Dog Walkers Weekly             1
                                                                        Epicurean Delights             1
                                                                        Iguana Lovers Monthly          1
                                                                        Monkey Madness                 1
                                                                        Your Health Tips               1
00025e267f294e03bc204fcdfe149139acb9e914bba8277284010050f15d694b  2015  Your Health Tips               1
000269f8bcff6cc09b39562d3d6fa572fafab3628f871ea30f40931e244e

In [217]:
pd.options.display.max_rows = 4000
data.groupby(['SubscriberID']).ListID.value_counts()


SubscriberID                                                      ListID                     
000079b501e0e82b7ef2addf1db8b860f6e3cb125d3b0b92fdb683f689cb5e68  Balkan News                    1
0001c6172c43da3dde25b87e10c23861250aed7ca2ce3c4d8eae1e41f51ef600  Fishing with Friends           1
0001cec76add203c3fea01001ed2729b0ddffaad9658b027307fe1932acd8771  Dog Walkers Weekly             1
                                                                  Epicurean Delights             1
                                                                  Iguana Lovers Monthly          1
                                                                  Monkey Madness                 1
                                                                  Your Health Tips               1
00025e267f294e03bc204fcdfe149139acb9e914bba8277284010050f15d694b  Your Health Tips               1
000269f8bcff6cc09b39562d3d6fa572fafab3628f871ea30f40931e244e25cd  Monkey Madness                 1
00034ffb62abfe7

In [145]:
#subscriberid and its corresponding number of lists
from collections import Counter
count=Counter(data.SubscriberID)
count

Counter({'b48af83aba984d79ef03905524b945ae15dffe9f799e7bf6e4d8272e35aacdc6': 1,
         'c52017dc1011082ffad85d6eedb8c4d0c11ac3aa066a7fd7aaa91778bbb119a2': 1,
         '35d6e429b1a1ffee51e55aa7e84bc6d8b6d26810875aabe5fe5d8ae89eabb67f': 3,
         '383fa5048195b358c4ca37996210822ef00b1badc7855e283712bc57fd1a04f4': 2,
         '1694a2d72442205b73b2ee7e3611cea0b586c9c2f8137ffc6ec921e79df2ed68': 1,
         'a8b0f46e752c533aa474eeab83301fee2083f38603d21b156154ac60b1bb51ad': 1,
         'b189c45a5ffe17dee6f24fcf6a9e08569f91ddaec31aa4e37727b79c53094b54': 2,
         '564df9b701870e6ae5f7753e89d37ca8cd5c35c9717fcd0f4fe80decf29e94c4': 2,
         'de24d3da8c7cbecce68f3d48b71f8cadb4f989fb5d7960b2fe37761dd6cfd8f6': 1,
         'd9d14fb6f2c2052655bcc81e8a90f72a5310371565e699ad5bc3ead847be05f2': 1,
         'cc8a247ebfe23a40ec82cfe28e3959c88d11c7c499b529dfdfb37cc55b72826e': 1,
         '18bca99e5ed95ded5782830a6657d2209d45d6f55b9769d87871bf13f79e4ad8': 1,
         '9367b301688e2674c9b17115870c3b

In [148]:
#to find nmber of subscribers who su
d = dict((k, v) for k, v in count.items() if v >= 2)

In [150]:
len(d)

28850

There are 28850 subscribers who subscribed for more than one list

In [23]:
# when only active subscribers are considered
combi_data.groupby(['SubscriberID']).ListID.value_counts()

SubscriberID                                                      ListID                    
000079b501e0e82b7ef2addf1db8b860f6e3cb125d3b0b92fdb683f689cb5e68  Balkan News                   1
0001c6172c43da3dde25b87e10c23861250aed7ca2ce3c4d8eae1e41f51ef600  Fishing with Friends          1
0001cec76add203c3fea01001ed2729b0ddffaad9658b027307fe1932acd8771  Dog Walkers Weekly            1
                                                                  Epicurean Delights            1
                                                                  Iguana Lovers Monthly         1
                                                                  Monkey Madness                1
                                                                  Your Health Tips              1
00025e267f294e03bc204fcdfe149139acb9e914bba8277284010050f15d694b  Your Health Tips              1
000269f8bcff6cc09b39562d3d6fa572fafab3628f871ea30f40931e244e25cd  Monkey Madness                1
00034ffb62abfe7689e129588

In [24]:
from collections import Counter
count2=Counter(combi_data.SubscriberID)
count2

Counter({'c52017dc1011082ffad85d6eedb8c4d0c11ac3aa066a7fd7aaa91778bbb119a2': 1,
         '35d6e429b1a1ffee51e55aa7e84bc6d8b6d26810875aabe5fe5d8ae89eabb67f': 3,
         '850bbb930deb965de5462e1e63db0f44e90ff9268fe9818792c4ce601574cdea': 1,
         '1694a2d72442205b73b2ee7e3611cea0b586c9c2f8137ffc6ec921e79df2ed68': 1,
         'a8b0f46e752c533aa474eeab83301fee2083f38603d21b156154ac60b1bb51ad': 1,
         'b189c45a5ffe17dee6f24fcf6a9e08569f91ddaec31aa4e37727b79c53094b54': 2,
         'de24d3da8c7cbecce68f3d48b71f8cadb4f989fb5d7960b2fe37761dd6cfd8f6': 1,
         'cc8a247ebfe23a40ec82cfe28e3959c88d11c7c499b529dfdfb37cc55b72826e': 1,
         '18bca99e5ed95ded5782830a6657d2209d45d6f55b9769d87871bf13f79e4ad8': 1,
         '9367b301688e2674c9b17115870c3b7e8b0931a29ce801e625729a683dd7c82e': 5,
         'dbc76221734ba8d4d4072f98e382b5f29af5e9b0c20f754ed2fbb2fdf24ddb5d': 2,
         '86c904a073c67451be2ed1df5260389e5691bcdfc5a8ee965abd53b8a5680ec2': 9,
         '35f045e02ab81dee04e419afb5a9ce

In [25]:
d1 = dict((k, v) for k, v in count2.items() if v >= 2)

In [26]:
len(d1)

20956

In [ ]:
Considering